MacOS build broken by commit 2663770 by @m9710797

@m9710797 committed [https://github.com/openMSX/openMSX/commit/26637700cb31ecb5b2c06fa7fd8eb745ca58c6e6](Change return type from 'const string&' to 'string_view' This patch changes the return type for (some) virtual methods that return 'const std::string&' to 'std::string_view'...) This commit broke the macOS build on clang:

In file included from src/PluggableFactory.cc:31:
src/serial/MidiOutCoreMIDI.hh:72:35: error: virtual function 'getName' has a different return type ('const std::string &' (aka 'const basic_string<char, char_traits<char>, allocator<char> > &')) than the function it overrides (which has return type 'std::string_view' (aka 'basic_string_view<char>'))
        [[nodiscard]] const std::string& getName() const override;
                            ~~~~~~~~~~~~ ^
src/Pluggable.hh:18:41: note: overridden virtual function is here
        [[nodiscard]] virtual std::string_view getName() const;
                              ~~~~~~~~~~~~~~~~ ^
1 error generated.
make[1]: *** [derived/x86_64-darwin-opt-3rd/obj/PluggableFactory.cc.o] Error 1
make: *** [staticbindist] Error 2

Not sure if this is because clang is more strict than gcc or something else.

--- a/src/serial/MidiOutCoreMIDI.hh
+++ b/src/serial/MidiOutCoreMIDI.hh
@@ -69,7 +69,7 @@ public:
 	// Pluggable
 	void plugHelper(Connector& connector, EmuTime::param time) override;
 	void unplugHelper(EmuTime::param time) override;
-	[[nodiscard]] const std::string& getName() const override;
+	[[nodiscard]] std::string_view getName() const override;
 	[[nodiscard]] std::string_view getDescription() const override;
 
 	// MidiOutMessageBuffer

